page.go

```diff
@@ -164,6 +164,15 @@ func (p *Page) ViewportE(params *proto.EmulationSetDeviceMetricsOverride) error
 	return err
 }
 
+// GetViewportE sets and returns the current viewport
+func (p *Page) GetViewportE() (*proto.PageVisualViewport, error) {
+	view, err := proto.PageGetLayoutMetrics{}.Call(p)
```

Comment:
After reading this, I afraid that `GetLayoutMetrics` is not equal to `setDeviceMetricsOverride`

The 15px error in your previous commits is the size of the scrollbar.

How about we change this code to pseudo code:

```go
func (p *Page) GetViewportE() (*proto.EmulationSetDeviceMetricsOverride, error) {
     client := p.Eval(`() => ({ w: innerWidth, h: innerHeight  })`)
     view, err := proto.PageGetLayoutMetrics{}.Call(p)
     return proto.EmulationSetDeviceMetricsOverride{client.w, client.h, scale: view.visualViewport.scale}
}
```

Reply:
Or just switch back to the `Page.viewport` private member. I know its ugly, but it seems to be the safest way for now.

It's a limitation of chrome protocol, they should provide a way for us to get the current override metrics.

Reply:
What if we'd use [VisualViewport](https://pkg.go.dev/github.com/ysmood/rod/lib/proto?tab=doc#PageVisualViewport) instead of the JS code? 

In theory, that should return the same data as we need for `EmulationSetDeviceMetricsOverride` and fix the issues with 15px error with scrollbar.

Reply:
`PageVisualViewport` sounds ok to me

Reply:
Ok, re-read the code and it's already there :D

```
// from visualViewport
// ClientWidth Width (CSS pixels), excludes scrollbar if present.
ClientWidth float64 `json:"clientWidth"`

// ClientHeight Height (CSS pixels), excludes scrollbar if present.
ClientHeight float64 `json:"clientHeight"`
```

In this case, if I'm right, we'd need to return `proto.EmulationSetDeviceMetricsOverride` with `ClientWidth`, `ClientHeight` and `Scale` from `VisualViewport`

Reply:
then in `GetFullScreenshot` we'd call this before, change the viewport to full layout width/height and by the end of the function, change the viewport back to return from `GetViewport` (didn't know why you wanted to use `defer` but now I know how you're thinking about it :) )

Reply:
defer is a way to make sure the recovery will happen.

```go
open()
defer close()

do_something()
```

vs

```go
open()

do_something()
close()
```

1. The later code: if do_something panics, close will not be called which will cause leaking.
2. The first one just looks better because open and close are a pair, and put them near to each other is easier for others to understand the code.

Reply:
I know how `defer` works, didn't know why you wanted to use it there tho haha

Reply:
maybe I misunderstand something, anyway, you can code the way you want. When I see your code I will understand. I make mistakes, and I'm open to ideas that correct me 😄

Reply:
that's the name of the game, you have the same mindset as I have - learn, learn, learn! :D

Reply:
BTW, I checked `PageVisualViewport` seems like it won't include scrollbar size, maybe a bug of chrome?